### PR TITLE
Abort virtual folder creation and warn if there is a name clash

### LINKED ISSUE #3905: VFolders: Abort creation and warn if there is a name clash
This is quickly provide a workaround for #3809 so 2.7.0 can be quickly released. A more long term proper fix must be provided for #3809.

### PR DESCRIPTION
**Pull request by unho**
Fixes #3905.

Test instructions:

- Craft a JSON with a vfolder whose name is the name of a directory located just below the vfolder location.
- Run `add_vfolders` with such JSON file and see if creation of that particular vfolder is aborted with a warning.

### CODE REVIEW

**Comment by unho** [comment]
@dwaynebailey Adjusted to address your concerns.

**Comment by dwaynebailey** [comment]
lgtm

**Comment by dwaynebailey** [comment]
@unho Isn't this a great time to actually write a test that validates what this issue is actually addressing?

**Comment by julen** [comment]
> @unho Isn't this a great time to actually write a test that validates what this issue is actually addressing?

This. :+1:

File: pootle/apps/virtualfolder/models.py [hunk 1]
```diff
@@ -355,6 +366,13 @@ def save(self, *args, **kwargs):
             self.directory.pootle_path
         )
 
+        # Abort if there is some clash with an existing directory.
+        if Directory.objects.filter(pootle_path=self.pootle_path).exists():
```
Comment (julen) [inline pootle/apps/virtualfolder/models.py]: This is pure field validation, and it's better off in `clean_fields()` as `VirtualFolder`s do. Also, no apparent need for custom exceptions, this is just `ValidationError`.

Comment (unho) [inline pootle/apps/virtualfolder/models.py]: Changed.

**Comment by julen** [comment]
Feel free to merge, although what @dwaynebailey said applies :dancer:

**Comment by unho** [comment]
Merged, but I am still working on adding some tests.

**Pull request merged**